Repo cant be used as a git depdendency

Not sure if the build script can be more generalized, but if you try to patch the repo and override tfmicro in your cargo toml with your git branch, it fails

```bash
error: failed to run custom build command for `tfmicro v0.1.0 (https://github.com/jacobrosenthal/tfmicro.git?branch=broken#e310699f)`

Caused by:
  process didn't exit successfully: `/home/jacob/Downloads/atsamd/boards/pygamer/target/release/build/tfmicro-14a54307ceb02d43/build-script-build` (exit code: 101)
  --- stdout
  Running bindgen
  Setting bindgen to cross compile to thumbv7em-none-eabihf
  OPT_LEVEL = Some("3")
  TARGET = Some("thumbv7em-none-eabihf")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_thumbv7em-none-eabihf = None
  CC_thumbv7em_none_eabihf = None
  TARGET_CC = None
  CC = None
  CROSS_COMPILE = None
  CFLAGS_thumbv7em-none-eabihf = None
  CFLAGS_thumbv7em_none_eabihf = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  CARGO_CFG_TARGET_FEATURE = Some("dsp,mclass,thumb-mode,v5te,v6,v6k,v6t2,v7")
  OPT_LEVEL = Some("3")
  TARGET = Some("thumbv7em-none-eabihf")
  HOST = Some("x86_64-unknown-linux-gnu")
  CXX_thumbv7em-none-eabihf = None
  CXX_thumbv7em_none_eabihf = None
  TARGET_CXX = None
  CXX = None
  CROSS_COMPILE = None
  CXXFLAGS_thumbv7em-none-eabihf = None
  CXXFLAGS_thumbv7em_none_eabihf = None
  TARGET_CXXFLAGS = None
  CXXFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  CARGO_CFG_TARGET_FEATURE = Some("dsp,mclass,thumb-mode,v5te,v6,v6k,v6t2,v7")

  --- stderr
  clang version 10.0.0-4ubuntu1 
  Target: thumbv7em-none-unknown-eabihf
  Thread model: posix
  InstalledDir: 
  ignoring nonexistent directory "/home/jacob/.cargo/git/checkouts/tfmicro-a2e3c4288cbcca64/e310699/submodules/tensorflow/tensorflow/lite/micro/tools/make/downloads/flatbuffers/include"
  ignoring nonexistent directory "/usr/share/gcc-arm-none-eabi-9-2020-q2-update/bin/../arm-none-eabi/include/c++/v1"
  ignoring nonexistent directory "lib/clang/10.0.0/include"
  ignoring nonexistent directory "/usr/lib/lib/clang/10.0.0/include"
  ignoring duplicate directory "/usr/share/gcc-arm-none-eabi-9-2020-q2-update/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/include"
    as it is a non-system directory that duplicates a system directory
  #include "..." search starts here:
  #include <...> search starts here:
   /usr/share/gcc-arm-none-eabi-9-2020-q2-update/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/include/c++/9.3.1
   /usr/share/gcc-arm-none-eabi-9-2020-q2-update/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/include/c++/9.3.1/arm-none-eabi/thumb/v7e-m+fp/hard
   /usr/share/gcc-arm-none-eabi-9-2020-q2-update/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/include/c++/9.3.1/backward
   /usr/share/gcc-arm-none-eabi-9-2020-q2-update/bin/../lib/gcc/arm-none-eabi/9.3.1/include
   /usr/share/gcc-arm-none-eabi-9-2020-q2-update/bin/../lib/gcc/arm-none-eabi/9.3.1/include-fixed
   /home/jacob/.cargo/git/checkouts/tfmicro-a2e3c4288cbcca64/e310699/submodules/tensorflow
   /usr/share/gcc-arm-none-eabi-9-2020-q2-update/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/include
  End of search list.
  /home/jacob/.cargo/git/checkouts/tfmicro-a2e3c4288cbcca64/e310699/submodules/tensorflow/tensorflow/lite/micro/micro_interpreter.h:21:10: fatal error: 'flatbuffers/flatbuffers.h' file not found
  /home/jacob/.cargo/git/checkouts/tfmicro-a2e3c4288cbcca64/e310699/submodules/tensorflow/tensorflow/lite/micro/micro_interpreter.h:21:10: fatal error: 'flatbuffers/flatbuffers.h' file not found, err: true
  thread 'main' panicked at 'Unable to generate bindings: ()', /home/jacob/.cargo/git/checkouts/tfmicro-a2e3c4288cbcca64/e310699/build.rs:382:33
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed

```

--- a/build.rs
+++ b/build.rs
@@ -17,6 +17,54 @@ use std::path::{Path, PathBuf};
 use std::process::Command;
 use std::time::Instant;
 
+use std::borrow::Borrow;
+use std::ffi::OsStr;
+
+fn run_command_or_fail<P, S>(dir: &str, cmd: P, args: &[S])
+where
+    P: AsRef<Path>,
+    S: Borrow<str> + AsRef<OsStr>,
+{
+    let cmd = cmd.as_ref();
+    let cmd = if cmd.components().count() > 1 && cmd.is_relative() {
+        // If `cmd` is a relative path (and not a bare command that should be
+        // looked up in PATH), absolutize it relative to `dir`, as otherwise the
+        // behavior of std::process::Command is undefined.
+        // https://github.com/rust-lang/rust/issues/37868
+        PathBuf::from(dir)
+            .join(cmd)
+            .canonicalize()
+            .expect("canonicalization failed")
+    } else {
+        PathBuf::from(cmd)
+    };
+    eprintln!(
+        "Running command: \"{} {}\" in dir: {}",
+        cmd.display(),
+        args.join(" "),
+        dir
+    );
+    let ret = Command::new(cmd).current_dir(dir).args(args).status();
+    match ret.map(|status| (status.success(), status.code())) {
+        Ok((true, _)) => return,
+        Ok((false, Some(c))) => panic!("Command failed with error code {}", c),
+        Ok((false, None)) => panic!("Command got killed"),
+        Err(e) => panic!("Command failed with error: {}", e),
+    }
+}
+
+fn check_submodules_or_checkout_from_git() {
+    if !Path::new("submodules/tensorflow/LICENSE").exists() {
+        eprintln!("Setting up submodules");
+        run_command_or_fail(".", "git", &["submodule", "update", "--init"]);
+    }
+    eprintln!("Done?");
+    if !Path::new("submodules/tensorflow/lite/micro/tools/make/downloads/flatbuffers/CONTRIBUTING.md").exists() {
+        eprintln!("Generating and fetching files using Sparkfun Edge target");
+        run_command_or_fail("submodules/tensorflow", "make", &["-f", "tensorflow/lite/micro/tools/make/Makefile", "TARGET=sparkfun_edge", "hello_world_bin"]);
+    }
+}
+
 fn manifest_dir() -> PathBuf {
     PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap())
 }
@@ -411,6 +459,7 @@ fn build_inline_cpp() {
 }
 
 fn main() {
+    check_submodules_or_checkout_from_git();
     bindgen_tflite_types();
     build_inline_cpp();
     cc_tensorflow_library();

--- a/build.rs
+++ b/build.rs
@@ -59,7 +59,7 @@ fn check_submodules_or_checkout_from_git() {
         run_command_or_fail(".", "git", &["submodule", "update", "--init"]);
     }
 
-    if !Path::new("submodules/tensorflow/lite/micro/tools/make/downloads/flatbuffers/CONTRIBUTING.md").exists() {
+    if !Path::new("submodules/tensorflow/tensorflow/lite/micro/tools/make/downloads/flatbuffers/CONTRIBUTING.md").exists() {
         eprintln!("Building tensorflow micro example to fetch Tensorflow dependencies");
         run_command_or_fail("submodules/tensorflow", "make", &["-f", "tensorflow/lite/micro/tools/make/Makefile", "test_micro_speech_test"]);
     }

--- a/build.rs
+++ b/build.rs
@@ -46,7 +46,7 @@ where
     );
     let ret = Command::new(cmd).current_dir(dir).args(args).status();
     match ret.map(|status| (status.success(), status.code())) {
-        Ok((true, _)) => return,
+        Ok((true, _)) => {}
         Ok((false, Some(c))) => panic!("Command failed with error code {}", c),
         Ok((false, None)) => panic!("Command got killed"),
         Err(e) => panic!("Command failed with error: {}", e),
@@ -58,10 +58,10 @@ fn check_submodules_or_checkout_from_git() {
         eprintln!("Setting up submodules");
         run_command_or_fail(".", "git", &["submodule", "update", "--init"]);
     }
-    eprintln!("Done?");
+
     if !Path::new("submodules/tensorflow/lite/micro/tools/make/downloads/flatbuffers/CONTRIBUTING.md").exists() {
-        eprintln!("Generating and fetching files using Sparkfun Edge target");
-        run_command_or_fail("submodules/tensorflow", "make", &["-f", "tensorflow/lite/micro/tools/make/Makefile", "TARGET=sparkfun_edge", "hello_world_bin"]);
+        eprintln!("Building tensorflow micro example to fetch Tensorflow dependencies");
+        run_command_or_fail("submodules/tensorflow", "make", &["-f", "tensorflow/lite/micro/tools/make/Makefile", "test_micro_speech_test"]);
     }
 }
 

